client/src/components/Messages.jsx

```diff
@@ -8,20 +8,70 @@ import {
   List,
   Fab,
 } from '@material-ui/core';
+import { useLocation } from 'react-router-dom';
 import Message from './Message.jsx';
 import UserContacted from './UserContacted.jsx';
 import { AuthContext } from '../contexts/index.jsx';
 import dummyData from '../../../messagesDummyData';
 
-function Messages({ userId }) {
-  const [usersContacted, setUsersContacted] = useState(dummyData.usersContacted);
+function Messages() {
+  const [usersContacted, setUsersContacted] = useState([]);
   const [messages, setMessages] = useState([]);
-  const [userContactedId, setUserContactedId] = useState('null');
-  const { currentUser, addMessage, getMessages } = useContext(AuthContext);
+  const [userContactedId, setUserContactedId] = useState(null);
+  const {
+    currentUser,
+    addMessage,
+    getMessages,
+    getProfile,
+    updateUsersContacted,
+  } = useContext(AuthContext);
+  const didMounted = useRef(false);
+  const location = useLocation();
+  const profile = location.state;
 
-  // const getUsersContacted = () => {
-  //   return false;
-  // };
+  useEffect(() => {
+    if (didMounted.current) {
+      if (profile) {
+        let userAlreadyContacted = false;
+        for (let i = 0; i < usersContacted.length; i += 1) {
+          if (usersContacted[i].userId === profile.userId) {
+            userAlreadyContacted = true;
+            break;
+          }
+        }
+        if (!userAlreadyContacted) {
+          const newContact = {
+            name: profile.name, userId: profile.userId, image: profile.image.url,
+          };
+          updateUsersContacted(currentUser.uid, newContact)
+            .then(() => {
+              const userContact = {
+                name: currentUser.displayName, userId: currentUser.uid, image: currentUser.photoURL,
+              };
+              updateUsersContacted(newContact.userId, userContact)
+                .then(() => {
+                  setUsersContacted(usersContacted => [...usersContacted, newContact]);
+                });
+            })
+            .catch(err => console.log(err));
+        }
+      }
+    } else {
+      didMounted.current = true;
```

Comment:
I'm being nitpicky again --I hope you don't mind. 'didMounted' could be changed to 'didMount' or 'alreadyMounted'